lib/authentication-context.js

```diff
@@ -142,6 +142,12 @@ Object.defineProperty(AuthenticationContext.prototype, 'options', {
   }
 });
 
+Object.defineProperty(AuthenticationContext.prototype, 'extraQP', {
```

Comment:
Sure, it makes sense, and will remove it. I saw in the android-adal and adal-js, they all have ways to specify extra query parameter sent to request. This could kinda avoid the slice=testslice mistake we saw in the master branch.